src/main/java/co/inventorsoft/StreamHomework.java

```diff
@@ -29,7 +34,19 @@
      * @return collection of user, without duplicates
      */
     public List<User> createUsers(final List<String> emails) {
-        return null;
+        List<String> emailsWithoutDup = emails.stream()
+                .filter(email-> email!=null && !email.isEmpty())
+                .distinct()
+                .collect(toList());
+
+        List<User> users = emailsWithoutDup.stream()
+                .map(p -> new User(p))
+                .collect(Collectors.toList());
```

Comment:
This is much better now!
Though, you can do it in single stream, you don't need 3 `.collect()` operators here.